requirements.txt

```diff
@@ -0,0 +1 @@
+deprecation>=2
```

Comment:
@csadorf In the words of @rosecers, this comment is a "signal-boost" that signac is no longer dependency-free. I'm not sure if that affects anything else beyond this repo (potentially CI?) and the conda-forge recipe.

Reply:
The added implicit dependencies are `packaging` and `pyparsing`, which are both *noarch*. So I'm ok with that. We have been using the language of "minimal dependencies" instead of "dependency free" for a while now.

Reply:
Is this discussion just about whether the file needs to exist?